### PR TITLE
fix(governance,networks): remove localhost from networks (again)

### LINKED ISSUE #13307: Do not publish `localhost` network in npm package
Our networks package on npm includes the `localhost` network. We should probably not include it :)

### PR DESCRIPTION
**Pull request by clemsos**
# Description

This adds a fix to remove `localhost` from our networks package 

# Issues

Fixes #13307
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: packages/networks/bin/localhost.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  const localhost = {
+    ...defaultLocalhost,
+    subgraph: {
+      endpoint: subgraphEnpoint,
+    },
+    unlockAddress,
+  }
+
+  // log for debug purposes
+  console.log(localhost)
```
Comment (julien51) [inline packages/networks/bin/localhost.js:40]: Do you wnat to keep that?

File: packages/networks/bin/localhost.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  const [networkInfoPath, subgraphEnpoint] = process.argv.slice(2)
+
+  const networkInfo = await fs.readJSON(networkInfoPath)
+  const {
+    localhost: {
+      Unlock: { address: unlockAddress },
+    },
+  } = networkInfo
+
+  console.log(networkInfo)
```
Comment (julien51) [inline packages/networks/bin/localhost.js:62]: same?

File: packages/networks/bin/localhost.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
+import fs from 'fs-extra'
```
Comment (julien51) [inline packages/networks/bin/localhost.js:1]: Can you add a quick note in the README how the `localhost` network is different from he other ones?

**Review by julien51 — APPROVED** [review]
Minor things to fix/clarify!

File: packages/networks/bin/localhost.js (hunk 1 above)
Comment (clemsos) [reply packages/networks/bin/localhost.js:40]: yes lets keep that one so we have it to debug on CI

File: packages/networks/bin/localhost.js (hunk 3 above)
Comment (clemsos) [reply packages/networks/bin/localhost.js:1]: yep added

File: packages/networks/bin/readSubgraphInfo.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+// this is used to read a network
+const run = async () => {
+  const [networkInfoPath] = process.argv.slice(2)
+  const networkInfo = await fs.readJSON(networkInfoPath)
+  const {
+    localhost: {
+      Unlock: { address: unlockAddress },
+    },
+  } = networkInfo
+  console.log(networkInfo)
```
Comment (clemsos) [inline packages/networks/bin/readSubgraphInfo.js:12]:
```suggestion
```

File: tests/package.json [hunk 5]
```diff
@@ -4,9 +4,9 @@
   "scripts": {
     "lint:fix": "yarn lint --fix",
     "lint": "eslint --resolve-plugins-relative-to ../eslint-config --ext .tsx,.ts,.js scripts/ test/",
-    "test": "yarn test:subgraph",
-    "test:subgraph": "hardhat test",
-    "ci": "hardhat test --network 0.0.0.0"
+    "test": "hardhat test",
+    "test:prepare": "sh bin/prepare.sh",
+    "ci": "yarn test:prepare && yarn test --network 0.0.0.0"
```
Comment (clemsos) [inline tests/package.json:9]:
```suggestion
    "ci": "yarn test:prepare && yarn test"
```

**Pull request merged**